docs/design/2022-05-24-delete-feature.md

```diff
@@ -0,0 +1,95 @@
+# Flink connector Delete Feature
+
+- Author(s): [Shi Yuhang](http://github.com/shiyuhang0)
+- Tracking Issue: https://github.com/tidb-incubator/TiBigData/issues/200
+
+## Table of Contents
+
+* [Introduction](#introduction)
+* [Motivation or Background](#motivation-or-background)
+* [Detailed Design](#detailed-design)
+  * [New Configuration](#new-configuration)
+  * [Main Steps](#main-steps)
+  * [Delete Logical](#delete-logical)
+  * [Row Order](#row-order)
+* [Compatibility](#compatibility)
+* [Test Design](#test-design)
+
+## Introduction
+
+Support delete feature in streaming mode for TiBigData/Flink.
+
+## Motivation or Background
+
+Currently, TiBigData/Flink doesn't support DELETE RowKind in the TiKV sink. In other words, we can't consume delete changelog to execute delete.
+As a real batch&streaming engine, it's necessary to support delete in Flink. 
+- Delete will bypass TiDB
+- Delete is supported in streaming mode
+
+## Detailed Design
+
+### New Configuration
+
+We introduce a new configuration `sink.tikv.delete-enable` to control delete.
+- The configuration is a boolean type with the default value `false`, which will disable the delete feature.
+- The configuration can only work in MINIBATCH transaction and upsert mode, or delete RowKind will be filtered.
+- Only support delete from table with pk/uk, make sure at least one pk/uk's value is not null. (every column should not be null for multiple-column pk/uk)
+
+### Main Steps
+
+Here are the main steps to support the delete feature:
+- Add the configuration to open delete.
+- Check if delete is enabled. If you are not in MINIBATCH transaction or upsert mode, delete will be disabled even you configure `sink.tikv.delete-enable` to `true`.
```

Comment:
Should we throw an exception to tell users `delete row is not supported in xxx mode`?

Reply:
i will throw an exception